Composer/packages/client/__tests__/components/Adapters/ABSChannels.test.tsx

```diff
@@ -0,0 +1,246 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import * as React from 'react';
+import { act, fireEvent } from '@botframework-composer/test-utils';
+
+import { renderWithRecoil } from '../../testUtils/renderWithRecoil';
+import ABSChannels from '../../../src/pages/botProject/adapters/ABSChannels';
+import { botDisplayNameState, settingsState } from '../../../src/recoilModel';
+import * as authUtils from '../../../src/utils/auth';
+import httpClient from '../../../src/utils/httpUtil';
+
+jest.mock('../../../src/utils/httpUtil');
+
+const mockNavigationTo = jest.fn();
+jest.mock('../../../src/utils/navigation', () => ({
+  navigateTo: (...args) => mockNavigationTo(...args),
+}));
+
+const CHANNELS = {
+  TEAMS: 'MsTeamsChannel',
+  WEBCHAT: 'WebChatChannel',
+  SPEECH: 'DirectLineSpeechChannel',
+};
+
+const mockProjId = '123';
+const mockDisplayName = '';
+const mockPublishTargetName = 'target1';
+const mockConfigName = mockPublishTargetName;
+const mockBotName = 'mockBotName';
+const mockAppId = '123';
+const mockSubscriptionId = '456';
+const mockTenantId = '123';
+const mockResourceGroup = 'mockResourceGroup';
+const mockTokenValue = '123';
+
+const mockTargetConfig = {
+  name: mockConfigName,
+  botName: mockBotName,
+  MicrosoftAppId: mockAppId,
+  resourceGroup: mockResourceGroup,
+  tenantId: mockTenantId,
+  subscriptionId: mockSubscriptionId,
+};
+
+const luisConfig = {
+  name: '',
+  authoringKey: '12345',
+  authoringEndpoint: 'testAuthoringEndpoint',
+  endpointKey: '12345',
+  endpoint: 'testEndpoint',
+  authoringRegion: 'westus',
+  defaultLanguage: 'en-us',
+  environment: 'composer',
+};
+
+const qnaConfig = { subscriptionKey: '12345', endpointKey: '12345', qnaRegion: 'westus' };
+
+const mockSettingsState = {
+  luis: luisConfig,
+  qna: qnaConfig,
+  defaultLanguage: 'en-us',
+  languages: ['en-us'],
+  luFeatures: {},
+  runtime: {
+    key: '',
+    customRuntime: true,
+    path: '',
+    command: '',
+  },
+  importedLibraries: [],
+  customFunctions: [],
+  publishTargets: [
+    {
+      name: mockPublishTargetName,
+      type: 'azurewebapp',
+      configuration: JSON.stringify(mockTargetConfig),
+    },
+  ],
+};
+
+describe('<ABSChannels />', () => {
+  beforeEach(() => {
+    (httpClient.get as jest.Mock) = jest
+      .fn()
+      .mockResolvedValue({ data: { properties: { properties: { acceptedTerms: true } } } });
+    (httpClient.put as jest.Mock) = jest.fn().mockResolvedValue({});
+    (httpClient.delete as jest.Mock) = jest.fn().mockResolvedValue({});
+
+    jest.spyOn(authUtils, 'isShowAuthDialog').mockReturnValue(false);
```

Comment:
some of this might need to be updated when the auth refactor lands.

Reply:
Yeah I figured that would be the case. Seeing as these are going in before the Auth change then these unit tests will actually break the CI build on your auth PR and block it from merging. I am finishing up the unit tests today and can pivot to your branch after to start refacoring them there to unblock the merge